pkg/model/imagemetadata.proto

```diff
@@ -0,0 +1,37 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package pipe.model;
+option go_package = "github.com/pipe-cd/pipe/pkg/model";
+
+import "validate/validate.proto";
+
+message ImageMetadata {
```

Comment:
I'm thinking about the model's name. I wonder `ImageMetadata` is a little bit versatile name. This is for only the latest image. It's a bit weird to fetch from here one when we want to have the latest image.
And I guess instead of Metadata, Manifest is more readable according to [here](https://docs.docker.com/registry/spec/api/#pulling-an-image).

Therefore, how about `LatestImage` or `LatestImageManifest`?

Any objection is welcome as I'm still debating.

Reply:
To be honest I was thinking about those 2 things too.

> About the naming `Metadata` or `Manifest`


When I was thinking which is better, I checked the definition of docker image manifest at here
https://docs.docker.com/registry/spec/manifest-v2-2/

and by running the following command I got a schema that does not fit our purpose.

```
docker manifest inspect docker.io/library/alpine:latest
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:074d3636ebda6dd446d0d00304c4454f468237fdacf08fb0eeac90bdbfa1bac7",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:096ebf69d65b5dcb3756fcfb053e6031a3935542f20cd7a8b7c59e1b3cb71558",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v6"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:299294be8699c1b323c137f972fd0aa5eaa4b95489c213091dcf46ef39b6c810",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:549694ea68340c26d1d85c00039aa11ad835be279bfd475ff4284b705f92c24e",
         "platform": {
            "architecture": "arm64",
            "os": "linux",
            "variant": "v8"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:a2c49065a4a5161095f5d60c527f8b8ab3d4a4e0af79333eeccf4dd260aae64d",
         "platform": {
            "architecture": "386",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:1f2e2eda6c64b2d73549fa66378f5c3474ba1fad4ef7ed94fd7298ca9e8862fd",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:4217ef60b4b11942f76d992798c5904782ebb86f70742800644da32ba3a6f25b",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}docker manifest inspect docker.io/library/alpine:latest
{
   "schemaVersion": 2,
   "mediaType": "application/vnd.docker.distribution.manifest.list.v2+json",
   "manifests": [
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:074d3636ebda6dd446d0d00304c4454f468237fdacf08fb0eeac90bdbfa1bac7",
         "platform": {
            "architecture": "amd64",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:096ebf69d65b5dcb3756fcfb053e6031a3935542f20cd7a8b7c59e1b3cb71558",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v6"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:299294be8699c1b323c137f972fd0aa5eaa4b95489c213091dcf46ef39b6c810",
         "platform": {
            "architecture": "arm",
            "os": "linux",
            "variant": "v7"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:549694ea68340c26d1d85c00039aa11ad835be279bfd475ff4284b705f92c24e",
         "platform": {
            "architecture": "arm64",
            "os": "linux",
            "variant": "v8"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:a2c49065a4a5161095f5d60c527f8b8ab3d4a4e0af79333eeccf4dd260aae64d",
         "platform": {
            "architecture": "386",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:1f2e2eda6c64b2d73549fa66378f5c3474ba1fad4ef7ed94fd7298ca9e8862fd",
         "platform": {
            "architecture": "ppc64le",
            "os": "linux"
         }
      },
      {
         "mediaType": "application/vnd.docker.distribution.manifest.v2+json",
         "size": 528,
         "digest": "sha256:4217ef60b4b11942f76d992798c5904782ebb86f70742800644da32ba3a6f25b",
         "platform": {
            "architecture": "s390x",
            "os": "linux"
         }
      }
   ]
}
```

Reply:
> About prefixing `Lastest` or not

I think the model name is fine because it represents the image entity (a specific entity may be the latest one or not).

But I was thinking about the name of DB collection. Firstly, I named the collection as `LatestImageMetadata` and the model is `ImageMetadata` but after all, I unified them. 

Happy to hear more of your thoughts?

Reply:
But after rethinking about the prefix, I realized that the struct is containing the `ID` field that was concatenated just by project id and repo name but not the tag. So prefixing the name with `Latest` looks better.

Reply:
Hmm... but this struct has not only reference but also repo name. I'm beginning to feel Metadata is enough.
(Also, there is already the `ImageRef` model.)

Reply:
>the struct is containing the ID field that was concatenated just by project id and repo name but not the tag. So prefixing the name with Latest looks better.

I feel the same way 😄

Reply:
(Btw, there are ImageRef and ImageName in the `model` package as mentioned before.)
https://github.com/pipe-cd/pipe/blob/ace3150fd61adddeaa8c1057c575e5b13b428404/pkg/model/image_name.go#L44-L54

Reply:
I agreed with your suggestion mentioned at https://github.com/pipe-cd/pipe/pull/1358#issuecomment-755189240, hence either `ImageMetadata` or `ImageReference` is our answer.

Reply:
I'm starting to feel "reference" is appropriate again. The "reference" clearly represents a specific image that has a unique digest. Wheres "metadata" seems to indicate immutable information about the repository.

Hmm... happy to hear your thought.

Reply:
I think "ImageMetadata" is still referring to the image, not to the repository.
But I'm also starting to get the feeling that "ImageReference" is better. (Also because of already existing `ImageRef`)
Let me update the name. 

Nice catch!